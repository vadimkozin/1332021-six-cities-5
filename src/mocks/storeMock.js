import configureStore from 'redux-mock-store';
import {SortingType, CITIES_NAMES, StatusType, AuthorizationStatus} from '../const';
import {offerAdapter, commentAdapter} from '../store/adapters';

import hotelJson from './json/hotel';
import hotelsNearbyJson from './json/hotels-nearby';
import offersJson from './json/offers';
import commentsJson from './json/comments';

const hotel = offerAdapter.adaptToClientOffer(hotelJson);
const hotelsNearby = offerAdapter.adaptToClient(hotelsNearbyJson);
const offers = offerAdapter.adaptToClient(offersJson);
const comments = commentAdapter.adaptToClient(commentsJson);

const initCity = CITIES_NAMES[0];

const Data = {
  SORT: `sort`,
  CITY: `city`,
  OFFERS: `offersGist`,
  HOTEL: `hotelGist`,
  HOTELS_NEARBY: `hotelsNearbyGist`,
  COMMENTS: `commentsGist`,
};

const basicState = {
  data: null,
  status: StatusType.IDLE,
  error: ``,
};

const makeInit = (basic = basicState) => (data = {}) => Object.assign({}, basic, {data});

const init = makeInit();

const storeConfig = configureStore();

const storeMock = storeConfig({
  data: {
    cities: CITIES_NAMES,
  },
  process: {
    [Data.CITY]: initCity,
    [Data.SORT]: SortingType.POPULAR,
    [Data.OFFERS]: init(offers),
    [Data.HOTEL]: init(hotel),
    [Data.HOTELS_NEARBY]: init(hotelsNearby),
    [Data.COMMENTS]: init(comments),
  },
  user: {
    id: 1,
    name: `qwerty`,
    email: `qwerty@mail.ru`,
    isPro: false,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  },
});

export {storeMock};
