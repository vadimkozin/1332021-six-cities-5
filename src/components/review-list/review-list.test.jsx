import React from 'react';
import {Provider} from 'react-redux';
import {render} from 'enzyme';
import storeMock, {comments} from '../../mocks/storeMock';
import ReviewList from './review-list';

describe(`ReviewList`, () => {
  it(`should render ReviewList component`, () => {
    const component = render(
        <Provider store={storeMock}>
          <ReviewList reviews={comments} isAuth={true}/>
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
