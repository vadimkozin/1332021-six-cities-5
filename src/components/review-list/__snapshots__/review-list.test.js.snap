// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList should render ReviewList component 1`] = `
<section
  class="property__reviews reviews"
>
  <h2
    class="reviews__title"
  >
    Reviews Â· 
    <span
      class="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    class="reviews__list"
  >
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Sophie
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width:60%"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)
        </p>
        <time
          class="reviews__time"
          datetime="2020-11-02"
        >
          November 2020
        </time>
      </div>
    </li>
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Mollie
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width:40%"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!
        </p>
        <time
          class="reviews__time"
          datetime="2020-11-02"
        >
          November 2020
        </time>
      </div>
    </li>
  </ul>
  <form
    action="#"
    class="reviews__form form"
    method="post"
  >
    <label
      class="reviews__label form__label"
      for="review"
    >
      Your review
    </label>
    <div
      class="reviews__rating-form form__rating"
    >
      <input
        class="form__rating-input visually-hidden"
        id="5-stars"
        name="rating"
        type="radio"
        value="5"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="5-stars"
        title="perfect"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            href="#icon-star"
          />
        </svg>
      </label>
      <input
        class="form__rating-input visually-hidden"
        id="4-stars"
        name="rating"
        type="radio"
        value="4"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="4-stars"
        title="good"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            href="#icon-star"
          />
        </svg>
      </label>
      <input
        class="form__rating-input visually-hidden"
        id="3-stars"
        name="rating"
        type="radio"
        value="3"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="3-stars"
        title="not bad"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            href="#icon-star"
          />
        </svg>
      </label>
      <input
        class="form__rating-input visually-hidden"
        id="2-stars"
        name="rating"
        type="radio"
        value="2"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="2-stars"
        title="badly"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            href="#icon-star"
          />
        </svg>
      </label>
      <input
        class="form__rating-input visually-hidden"
        id="1-stars"
        name="rating"
        type="radio"
        value="1"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="1-stars"
        title="terribly"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            href="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      class="reviews__textarea form__textarea"
      id="review"
      maxlength="300"
      minlength="50"
      name="review"
      placeholder="Tell how was your stay, what you like and what can be improved"
    />
    <div
      class="reviews__button-wrapper"
    >
      <p
        class="reviews__help"
      >
        To submit review please make sure to set 
        <span
          class="reviews__star"
        >
          rating
        </span>
         and describe your stay with at least 
        <b
          class="reviews__text-amount"
        >
          50 characters
        </b>
        .
      </p>
      <button
        class="reviews__submit form__submit button"
        disabled=""
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;
